api/src/org/apache/cloudstack/api/command/user/affinitygroup/UpdateVMAffinityGroupCmd.java

```diff
@@ -96,6 +96,10 @@ public Long getId() {
             throw new InvalidParameterValueException("affinitygroupids parameter is mutually exclusive with affinitygroupnames parameter");
         }
 
+        if (affinityGroupNameList == null && affinityGroupIdList == null) {
+            throw new InvalidParameterValueException("affinitygroupids parameter or affinitygroupnames parameter must be given");
+        }
+
```

Comment:
Lines 95-102 should be at the beginning of the `execute` method rather than in an accessor.